src/bufr/BufrParser/Exports/Variables/DatetimeVariable.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * (C) Copyright 2020 NOAA/NWS/NCEP/EMC
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#include <iostream>
+#include <ostream>
+#include <iomanip>
+
+#include "DatetimeVariable.h"
+
+
+namespace
+{
+    namespace ConfKeys
+    {
+        const char* Year = "year";
+        const char* Month = "month";
+        const char* Day = "day";
+        const char* Hour = "hour";
+        const char* Minute = "minute";
+        const char* Second = "second";
+        const char* Utc = "isUTC";
+    }  // namespace ConfKeys
+}  // namespace
+
+
+namespace Ingester
+{
+    DatetimeVariable::DatetimeVariable(const eckit::Configuration& conf) :
+      yearKey_(conf.getString(ConfKeys::Year)),
+      monthKey_(conf.getString(ConfKeys::Month)),
+      dayKey_(conf.getString(ConfKeys::Day)),
+      hourKey_(conf.getString(ConfKeys::Hour)),
+      minuteKey_(conf.getString(ConfKeys::Minute)),
+      secondKey_(conf.getString(ConfKeys::Second)),
+      isUTC_(conf.getBool(ConfKeys::Utc))
+    {
+    }
+
+    std::shared_ptr<DataObject> DatetimeVariable::exportData(const BufrDataMap& map)
+    {
+        auto datetimes = std::vector<std::string>();
+
+        datetimes.reserve(map.at(yearKey_).size());
+        for (unsigned int idx = 0; idx < map.at(yearKey_).size(); idx++)
+        {
+            // YYYY-MM-DDThh:mm:ssZ
+            std::ostringstream datetimeStr;
+            datetimeStr << std::setfill('0')
+                        << std::setw(4) << map.at(yearKey_)(idx) << "-" \
+                        << std::setw(2) << map.at(monthKey_)(idx) << "-" \
+                        << std::setw(2) << map.at(dayKey_)(idx) << "T" \
+                        << std::setw(2) << map.at(hourKey_)(idx) << ":" \
+                        << std::setw(2) << map.at(minuteKey_)(idx) << ":" \
+                        << std::setw(2) << map.at(secondKey_)(idx);
+
+            if (isUTC_)
```

Comment:
Not needed for this PR, but suggestion:
if not isUTC_, provide a time offset in YAML and convert to UTC time. Ex. UTC_offset=-5, to convert obs in EST to UTC.

Reply:
I went ahead and fixed these because I don't want a lot of YAML files being generated with out of date syntax. I added a new parameter called "hoursFromUtc" as an optional and I've deprocated "isUtc". By default hoursFromUtc is 0. I also made the parameter "second" optional while I was at it. I discovered I'd inadvertently created duplicates for this class (DatetimeExport) so I removed that and MnemonicExport (also a duplicate).